src/Services.Remoting/Client/OmexServiceProxyFactory.cs

```diff
@@ -11,12 +12,30 @@ namespace Microsoft.Omex.Extensions.Services.Remoting.Client
 	/// </summary>
 	public static class OmexServiceProxyFactory
 	{
-		/// <summary>
-		/// Instance of the ServiceProxyFactory
-		/// </summary>
-		public static ServiceProxyFactory Instance { get; } = new ServiceProxyFactory(handler =>
+		private static ServiceProxyFactory s_serviceProxyFactory = new ServiceProxyFactory(handler =>
 			new OmexServiceRemotingClientFactory(
 				new FabricTransportServiceRemotingClientFactory(
 					remotingCallbackMessageHandler: handler)));
+
+		/// <summary>
+		/// Binds transport settings to the service proxy factory, for use in secure remoting communication.
+		/// </summary>
+		public static void WithTransportSettings(FabricTransportRemotingSettings transportSettings)
```

Comment:
Usage will be awkward if you need to make remoting calls from several places. It might make more sense to implement IServiceProxyFactory and register in DI. Static insecure instance can be preserved for compatibility and obsoleted later. It would be good to review the whole remoting approach as well, see what is the value added and what we can potentially improve.